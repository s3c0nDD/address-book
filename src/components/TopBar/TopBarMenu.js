import React, { useMemo } from 'react';
import { Link } from 'react-router-dom';

import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';

import { ROUTES } from '../../Routes';

const TopBar = (props) => {
  const { anchorEl, handleClose, isOpen } = props;

  const originPositionObj = useMemo(() => ({
    vertical: 'top',
    horizontal: 'left',
  }), []);

  return (
    <Menu
      anchorEl={anchorEl}
      anchorOrigin={originPositionObj}
      keepMounted
      transformOrigin={originPositionObj}
      open={isOpen}
      onClose={handleClose}
    >
      <MenuItem
        onClick={handleClose}
        component={Link}
        to={ROUTES.root}
      >
        Users Gallery
      </MenuItem>
      <MenuItem
        onClick={handleClose}
        component={Link}
        to={ROUTES.settings}
      >
        Settings
      </MenuItem>
    </Menu>
  )
};

export default TopBar;