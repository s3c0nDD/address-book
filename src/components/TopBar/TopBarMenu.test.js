import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import TopBarMenu from './TopBarMenu';

describe('TopBarMenu', () => {
  let shallow;
  let component;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    const anchorEl = (<div className="anchorElement" />);
    component = (
      <TopBarMenu
        anchorEl={anchorEl}
        handleClose={jest.fn()}
        open={Boolean(anchorEl)}
      />
    );
    wrapper = shallow(component);
  });

  it('shallowly renders and matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
