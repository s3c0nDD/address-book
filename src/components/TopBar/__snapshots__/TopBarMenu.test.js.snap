// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarMenu shallowly renders and matches snapshot 1`] = `
<WithStyles(ForwardRef(Menu))
  anchorEl={
    <div
      className="anchorElement"
    />
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "top",
    }
  }
  keepMounted={true}
  onClose={[MockFunction]}
  open={true}
  transformOrigin={
    Object {
      "horizontal": "left",
      "vertical": "top",
    }
  }
>
  <WithStyles(ForwardRef(MenuItem))
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Link",
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    onClick={[MockFunction]}
    to="/"
  >
    Users Gallery
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Link",
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    onClick={[MockFunction]}
    to="/settings"
  >
    Settings
  </WithStyles(ForwardRef(MenuItem))>
</WithStyles(ForwardRef(Menu))>
`;
